### PR TITLE
HTTP: Rework interfaces to fixed 8080 and 8081

### PR DESCRIPTION
**Pull request by reinkrul**
Fixes https://github.com/nuts-foundation/nuts-node/issues/2860

Changes:
- Kept `MultiEcho` type to easily map specific endpoints (`/internal`, `/status`, `/metrics`) to a specific Echo server (`8081`). But instead of allowing mapping endpoints to an Echo server, you can only configure the port/address of the 2 HTTP interfaces (internal and external).
  - had to  
- Removed support for HTTPS on the server-side; parties must deploy a reverse proxy that handles TLS
- Moved HTTP logging configuration to the module-level config, now applies all HTTP interfaces
- API authentication now only applies to `/internal`
- Removed support for CORS: only used by IRMA/EmployeeID which will be removed at some point. Parties that need it can set the headers in their reverse proxy.

Most changes are in the e2e tests, because:
- `/internal` is bound to `127.0.0.1:8081` by default, to make it available from outside the container it has to be remapped to `:8081`
- some e2e tests require HTTPS (in particular the new OpenID4VC tests), which isn't supported by the Nuts node any more, so I had to put an NGINX proxy before it
- the old default port `1323` is now split into `8081` and `8080`, so if a test used both external and internal endpoints test actions needed to be rewritten.

TODO:
- [x] Fix e2e tests
- [x] Rewrite documentation

### CODE REVIEW

**Comment by reinkrul** [comment]
There's some documentation left regarding recommended deployment, which should be rewritten altogether. I left that largely untouched in this PR.

File: docs/pages/deployment/production-configuration.rst [hunk 1]
```diff
@@ -31,61 +27,37 @@ You can determine the intended public by observing the first part of the URL:
...
-
-In some deployments CORS can be required for the public IRMA authentication endpoints when the user-facing
-authentication page is hosted on a (sub)domain that differs from Nuts Node's IRMA backend. CORS can be enabled on a
-specific HTTP interface by specifying the domains allowed to make CORS requests as ``cors.origin`` (see the example above).
-Although you can enable CORS on the default endpoint it's not advised to do so in a production environment,
-because CORS itself opens up new attack vectors on node administrators.
+      # The following is the default configuration of the HTTP interfaces.
+      internal:
+        address: 127.0.0.1:8081
+        auth:
```
Comment (reinkrul) [inline docs/pages/deployment/production-configuration.rst:38]:
```suggestion
        # This enables API authentication on /internal endpoints
        auth:
```

File: docs/pages/getting-started/1-running-docker.rst [hunk 2]
```diff
@@ -27,7 +27,7 @@ If you want to run without Docker Compose you can use the following command from
 
 .. code-block:: shell
 
-  docker run --name nuts -p 1323:1323 \
+  docker run --name nuts -p 5555:5555 -p 8080:8080 -p 8081:8081 \
```
Comment (reinkrul) [inline docs/pages/getting-started/1-running-docker.rst:30]:
```suggestion
  docker run --name nuts -p 8080:8080 -p 8081:8081 \
```

File: docs/pages/getting-started/1-running-docker.rst [hunk 3]
```diff
@@ -61,7 +61,9 @@ Copy the following YAML file and save it as ``docker-compose.yaml`` in the worki
       environment:
         NUTS_CONFIGFILE: /opt/nuts/nuts.yaml
       ports:
-        - 1323:1323
+        - 5555:5555
```
Comment (reinkrul) [inline docs/pages/getting-started/1-running-docker.rst:64]:
```suggestion
```

File: docs/pages/release_notes.rst [hunk 4]
```diff
@@ -3,6 +3,46 @@
 Release notes
 #############
 
+************************
+TBD (v6.0.0)
```
Comment (woutslakhorst) [inline docs/pages/release_notes.rst:7]: Peanut

File: docs/pages/deployment/production-configuration.rst [hunk 5]
```diff
@@ -31,61 +27,38 @@ You can determine the intended public by observing the first part of the URL:
...
-because CORS itself opens up new attack vectors on node administrators.
+      # The following is the default configuration of the HTTP interfaces.
+      internal:
+        address: 127.0.0.1:8081
+        # This enables API authentication on /internal endpoints
+        auth:
+          type: token_v2
+          authorizedkeyspath: /opt/nuts/authorized_keys
+          audience: nuts-node.example.com
+      external:
```
Comment (woutslakhorst) [inline docs/pages/deployment/production-configuration.rst:43]: maybe public is better to signal internet accessibility?

File: docs/pages/deployment/production-configuration.rst [hunk 6]
```diff
@@ -31,61 +27,38 @@ You can determine the intended public by observing the first part of the URL:
...
+        address: :8080
 
 Authentication
 ^^^^^^^^^^^^^^
-You should secure some HTTP endpoints (notably ``/internal``) with authentication.
-The Nuts node supports JWT Bearer Token authentication which can be enabled on any HTTP endpoint.
-You can enable it by setting ``auth.type`` (see example above) to ``token_v2`` for the specific bind.
-The specified path, and all its subpaths, will then require a JWT token generated by an external
+
+You should secure access to ``/internal`` API endpoints () with authentication.
```
Comment (woutslakhorst) [inline docs/pages/deployment/production-configuration.rst:49]: leftover `()`

File: docs/pages/deployment/recommended-deployment.rst [hunk 7]
```diff
@@ -38,82 +38,82 @@ Nuts Node
...
+* **HTTP /iam**: for accessing OAuth2 and OpenID services.
 
-  *Users*: operators, SPs administrative and EHR applications.
+  *Users*: Other Nuts nodes, Verifiable Credential issuers and verifiers.
 
-  *Security*: restrict access through network separation and platform authentication.
+  *Security*: HTTPS with **publicly trusted** server certificate (on proxy). Monitor traffic to detect attacks.
 
-* **HTTP /public**: for accessing public services, e.g. IRMA authentication.
+* **HTTP /iam**: for accessing OAuth2 and OpenID services.
```
Comment (woutslakhorst) [inline docs/pages/deployment/recommended-deployment.rst:52]: duplicate entry?

**Review by woutslakhorst — CHANGES_REQUESTED** [review]

**Comment by woutslakhorst** [comment]
Also see #2857

File: docs/pages/release_notes.rst (hunk 4 above)
Comment (reinkrul) [reply docs/pages/release_notes.rst:7]: A peanut is botanically speaking not a nut, but okay..

File: docs/pages/deployment/production-configuration.rst (hunk 5 above)
Comment (reinkrul) [reply docs/pages/deployment/production-configuration.rst:43]: Yes, I like it better. I still don't like "internal" much, ideas?

File: docs/pages/release_notes.rst (hunk 4 above)
Comment (woutslakhorst) [reply docs/pages/release_notes.rst]: We focus on what you can buy at the Nuts stand.

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**